internal/docker/client.go

```diff
@@ -48,10 +50,50 @@ func GetClient() (*client.Client, error) {
 		}
 	}
 
-	dockerClient, err := client.NewClientWithOpts(clientOpts...)
+	possibleSocketPaths := possibleSocketPaths(runtime.GOOS)
+	for _, socketPath := range possibleSocketPaths {
+		dockerClient, err := newClient(socketPath, clientOpts...)
+		if err == nil {
+			err := checkConnection(dockerClient)
+			if err == nil {
+				return dockerClient, nil // Successfully connected
+			}
+		}
+	}
+
+	// If both attempts failed
+	return nil, fmt.Errorf("failed to connect to Docker daemon. Ensure Docker is running and accessible")
+}
+
+func checkConnection(dockerClient *client.Client) error {
+	ctx := context.Background()
+	_, err := dockerClient.Ping(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed create docker client: %w", err)
+		return fmt.Errorf("failed to ping Docker daemon: %w", err)
 	}
+	return nil
+}
 
-	return dockerClient, nil
+func newClient(socket string, opts ...client.Opt) (*client.Client, error) {
+	if socket == "" {
+		return client.NewClientWithOpts(opts...)
+	}
+	opts = append(opts, client.WithHost(socket))
+	return client.NewClientWithOpts(opts...)
+}
+
+func possibleSocketPaths(os string) []string {
+	switch os {
+	case "darwin":
+		hDir, err := homedir.Dir()
+		if err != nil {
+			return nil
```

Comment:
Should this return `[]string{""}` instead? Right now an error fetching the home directory will prevent the attempt to use the default/fromEnv socket on macOS.